### PR TITLE
feat(network_prov): Add wifi_prov or thread_prov in provision capabilities (IEC-105)

### PR DESCRIPTION
**Pull request by wqx6**
# Checklist

- [x] Component contains License
- [x] Component contains README.md
- [x] Component contains idf_component.yml file with `url` field defined
- [x] Component was added to [upload job](https://github.com/espressif/idf-extra-components/blob/master/.github/workflows/upload_component.yml#L18)
- [x] Component was added to [build job](https://github.com/espressif/idf-extra-components/blob/master/test_app/CMakeLists.txt#L8)
- [x] CI passing

# Change description
Add `wifi_prov` or `thread_prov` in provision capabilities for the provisioner to distinguish Thread or Wi-Fi devices.

### CODE REVIEW

**Review by chshu — APPROVED** [review]

File: network_provisioning/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+# 23-April-2024
+
+- Add `wifi_prov` or `thread_prov` in provision capabilities for the provisioner to distinguish Thread or Wi-Fi devices.
```
Comment (AdityaHPatwardhan) [inline network_provisioning/CHANGELOG.md:3]:
```suggestion
- Add `wifi_prov` or `thread_prov` in provision capabilities in the network provisioning manager for the provisioner to distinguish between Thread or Wi-Fi devices.
```

**Review by AdityaHPatwardhan — APPROVED** [review]

File: network_provisioning/examples/thread_prov/main/app_main.c [hunk 2]
```diff
@@ -112,7 +112,7 @@ static esp_err_t example_get_sec2_verifier(const char **verifier, uint16_t *veri
 const int THREAD_ATTACHED_EVENT = BIT0;
 static EventGroupHandle_t thread_event_group;
 
-#define PROV_QR_VERSION         "v2"
+#define PROV_QR_VERSION         "v1"
```
Comment (AdityaHPatwardhan) [inline network_provisioning/examples/thread_prov/main/app_main.c:115]: What is the requirement for this change?

Comment (AdityaHPatwardhan) [reply network_provisioning/examples/thread_prov/main/app_main.c:115]: Would this be considered as a breaking change?

Comment (wqx6) [reply network_provisioning/examples/thread_prov/main/app_main.c:115]: Previously I want to use the QR code to distinguish the Thread device and Wi-Fi device, so I added a network_type parameter in QR code and increase the QR code version, but now we can use the capabilities to distinguish the Thread device and Wi-Fi device.

The QRcode of network provisioning or wifi provisioning will be the same. And the phone app can use the same step to establish the secure session with end devices. 
 
After that the phone app can read the provision version string to identify when the device is a network-prov device or a wifi-prov device. (network-prov device's version string will start with a prefix netprov)
 
For wifi-prov device, the phone app should use previous wifi-prov protobuf to commission the device to a Wi-Fi network.
 
For network-prov device, the phone app should read the capabilities to identify the device is a Wi-Fi device or a Thread device, and send the Wi-Fi/Thread credential with the new network-provision protobuf.

**Review by mahavirj — APPROVED** [review]

**Pull request merged**